torchtune/modules/transformer.py

```diff
@@ -233,6 +399,15 @@ def forward(
             - d: embed dim
             - m_s: max seq len
         """
+        # input tensor of shape [b, s]
+        bsz, seq_len = tokens.shape
+
+        if seq_len > self.max_seq_len:
```

Comment:
I've hit this many times, but it's a bit confusing because the current suggestion is to set max_seq_len in the tokenizer. In the llama models the max_seq_len in the TransformerDecoder is hardcoded to the max for the model. Maybe we should make it more clear that the max_seq_len here is the limit for the model (if we want to enforce a limit) rather than what is actually truncated - probably as a separate follow up

Reply:
This becomes an issue when we look at kv_cache as the true max_seq_len might require much more memory allocated than we plan to use. On the other hand, this check should primarily check that the sequence isn't longer than what the model was trained for. We could expose a parameter in our builders for max_seq_len_cached or something, but I think that's out of scope of this work.